glean-core/python/glean/_dispatcher.py

```diff
@@ -7,23 +7,31 @@
 from typing import Callable, List, Tuple
 
 
-class DispatcherInternal:
+class Dispatcher:
```

Comment:
This is to change how the singleton of this is managed.  If you do something like `FooSingleton = Foo()` in Python, and then want to recreate (and thus reset the singleton), that doesn't affect all of the places that have already imported the singleton, so this was causing strange issues.  Moving this to a class with only `@classmethod` (static methods) seems like a safer pattern for singletons in Python.  (The `Glean` singleton already works this way).

Reply:
Oh. TIL.